staging/src/k8s.io/kubectl/pkg/cmd/clusterinfo/clusterinfo_dump.go

```diff
@@ -181,7 +172,33 @@ func (o *ClusterInfoDumpOptions) Run() error {
 		}
 	}
 
-	if err := o.PrintObj(nodes, setupOutputWriter(o.OutputDir, o.Out, "nodes", fileExtension)); err != nil {
+	// When writing json to stdout, we need to wrap the entire output in []
+	if writingToStdout && *o.PrintFlags.OutputFormat == "json" {
+		o.Out.Write([]byte("["))
+		defer o.Out.Write([]byte("]"))
+	}
+
+	separatorNeeded := false
+
+	dump := func(object runtime.Object, outputPath string) error {
```

Comment:
Here the same problem, the output should be consistent so I'm not sure why that is needed.

Reply:
See above, same reason, to make an array out of multiple JSON responses.  I can remove if we are OK with concatenated JSON fragments that are not in an outer array.